Content.Server/Psionics/PsionicsSystem.cs

```diff
@@ -76,13 +73,16 @@ private void OnMeleeHit(EntityUid uid, AntiPsionicWeaponComponent component, Mel
                     return;
                 }
 
-                if (component.Punish && HasComp<PotentialPsionicComponent>(entity) && !HasComp<PsionicComponent>(entity) && _random.Prob(0.5f))
+                if (component.Punish && !HasComp<PsionicComponent>(entity) && _random.Prob(0.5f))
```

Comment:
Like with https://github.com/Simple-Station/Einstein-Engines/pull/731/files#r1718973998, it may be a good idea to turn these into consts to make them easier to find when they require change